eogrow/utils/testing.py

```diff
@@ -91,24 +92,27 @@ def calculate_statistics(folder: str, config: StatCalcConfig) -> JsonDict:
         elif content_path.endswith((".geojson", ".gpkg")):
             stats[content] = _calculate_vector_stats(gpd.read_file(content_path), config)
         elif content_path.endswith(".parquet"):
-            try:
-                data = gpd.read_parquet(content_path)
-            except Exception:
-                data = _load_as_geoparquet(content_path)
-            stats[content] = _calculate_vector_stats(data, config)
+            stats[content] = _get_parquet_stats(content_path, config)
         else:
             stats[content] = None
 
     return stats
 
 
-def _load_as_geoparquet(path: str) -> gpd.GeoDataFrame:
-    data = pd.read_parquet(path)
-    if isinstance(data.geometry.iloc[0], str):
-        data.geometry = data.geometry.apply(wkt.loads)
-    elif isinstance(data.geometry.iloc[0], bytes):
-        data.geometry = data.geometry.apply(wkb.loads)
-    return gpd.GeoDataFrame(data, geometry="geometry", crs=data.utm_crs.iloc[0])
+def _get_parquet_stats(content_path: str, config: StatCalcConfig) -> JsonDict:
+    try:
+        data = gpd.read_parquet(content_path)
+        return _calculate_vector_stats(data, config)
+    except Exception:
+        data = pd.read_parquet(content_path)
+        if config.parquet_epsg:
+            data.geometry = data.geometry.apply(wkb.loads)
```

Comment:
should we do similar checks as below also here, or does the logic/assumption threshold change due to the presence of the `parquet_epsg`?

Reply:
in that case we could do a check whether we need to use wkb or wkt, but its starting to feel very convoluted...

Reply:
it still didn't feel completely right to me, how would you feel about the following:

```python
def _get_parquet_stats(content_path: str, config: StatCalcConfig) -> JsonDict:
    try:
        data = gpd.read_parquet(content_path)
        return _calculate_vector_stats(data, config)
    except Exception:
        data = pd.read_parquet(content_path)
        if "geometry" not in data:
            return _calculate_parquet_stats(data, config)

        loader = wkb.loads if isinstance(data.geometry.iloc[0], bytes) else wkt.loads
        data.geometry = data.geometry.apply(loader)

        if config.parquet_epsg:
            parsed_data = gpd.GeoDataFrame(data, geometry="geometry", crs=config.parquet_epsg)
            return _calculate_vector_stats(parsed_data, config)

        data.geometry = data.geometry.apply(wkt.dumps)
        return _calculate_parquet_stats(data, config)
```